src/platform/forms-system/src/js/containers/FormPage.jsx

```diff
@@ -92,13 +93,22 @@ class FormPage extends React.Component {
       : this.props.form.data;
   };
 
-  goBack = () => {
+  goBack = (customPath = '') => {
     const {
       form,
       route: { pageList },
       location,
     } = this.props;
-    const path = getPreviousPagePath(pageList, form.data, location.pathname);
+
+    const validCustomPath = customPath
+      ? getActiveExpandedPages(pageList, this.props.form.data).some(
+          page => page.path === customPath,
+        )
```

Comment:
If possible, could you break this logic out into a separate function? Either here or, if more appropriate, in the `routing` file? Just to make it clear what we're doing here.